src/table_plane_extractor_srv.py

```diff
@@ -13,9 +13,9 @@
 def transformPointCloud(cloud, target_frame, source_frame, tf_buffer):
```

Comment:
It was there before, but I believe it would be faster to extract planes and check the plane orientation instead of computing the normals and removing the non vertical points.

There is also an issue in the plane cloud extraction: there is nothing ensuring that only the main block of the plane is extracted as far as I can see. If parts of the shelf is in view for example, this would include points from the shelf that fits the plane model. This is mitigated by the normal based filtering, but you could still have two horizontal planes at the same height from disconnected furnitures, and you would rather have them as two separate planes then

Reply:
Implemented.